### ISSUE TITLE
autodoc cannot parse class/instance attributes type annotations

### ISSUE THREAD

**Issue by iwanb** [sphinx-doc/sphinx#4056]
### Problem
Using type annotation at the class level or in the `__init__` function breaks the automatic finding of class attributes.

#### Procedure to reproduce the problem
For example with this code:
```
class MyClass(object):
    "Hey"
    def __init__(self):
        self.attr = 'something'
        """My docstring."""

class OtherClass(object):
    "Ho"
    def __init__(self):
        self.attr: str = 'something'
        """My docstring."""

```

#### Error logs / results
MyClass is properly documented but OtherClass does not show any attribute.

If I try to force the class member, it gives a warning:
```
[autodoc] from todoc import OtherClass
[autodoc] import todoc
[autodoc] => <module 'todoc' from '/tmp/autodoc/todoc.py'>
[autodoc] getattr(_, 'OtherClass')
[autodoc] => <class 'todoc.OtherClass'>
[app] emitting event: 'autodoc-process-docstring'('class', 'todoc.OtherClass', <class 'todoc.OtherClass'>, {'members': ['attr']}, ['Ho', ''])
[autodoc] output:

.. py:class:: OtherClass()
   :module: todoc

   Ho
   
/tmp/autodoc/index.rst:16: (WARNING/2) missing attribute attr in object todoc.OtherClass [filtered system message]
[app] emitting event: 'doctree-read'(<document: <comment...><section "welcome to test's documentation!". ...>,)

[app] emitting event: 'env-updated'(<sphinx.environment.BuildEnvironment object at 0x7f3e0616e278>,)
/tmp/autodoc/index.rst:16: WARNING: missing attribute attr in object todoc.OtherClass

```

#### Expected results
I expect the attributes to be parsed the same whether there are type annotations or not.

### Reproducible project / your project
[autodoc.zip](https://github.com/sphinx-doc/sphinx/files/1302503/autodoc.zip)

### Environment info
- OS: Linux (ArchLinux)
- Python version: 3.6
- Sphinx version: lastest from master

**Comment by tk0miya** [comment]
I think this is already fixed by #4813. And it was released as 1.7.3.
Please try it out.
Thanks,

**Comment by iwanb** [comment]
Thank you!
